client/me/help/help-contact/index.jsx

```diff
@@ -103,7 +103,7 @@ class HelpContact extends Component {
 	backToHelp = () => {
 		const searchParams = new URLSearchParams( window.location.search );
 		const redirectPath = searchParams.get( 'redirect_to' ) ?? '';
-		if ( redirectPath.match( /^\/[^/]?/ ) ) {
+		if ( redirectPath.match( /^\/(?!\/)/ ) ) {
```

Comment:
This regexp match is validating whether the `redirect_to` URL is valid, e.g., doesn't contain some malicious `javascript:` snippet or redirect to some completely different site? And does that by checking if the path starts with `/` but doesn't start with `//`?

There's a better, more general-purpose way to do this check:
```js
function check( url ) {
  try {
    return new URL( url, window.location.href ).origin === window.location.origin;
  } catch {
    return false;
  }
}
```
It checks whether the link is a valid URL that is same-origin, i.e., inside the same site, and accepts also relative links like `help` or `./help` or `../help`, ones which would work well with `window.location.href = link`.

This could be extracted a to a common `lib` helper, as there are plenty of usage opportunites in Calypso. And making redirect validation as easy as possible looks like a great idea.